mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/gridfs/GridFSTest.java

```diff
@@ -0,0 +1,197 @@
+ */
+package org.hibernate.ogm.datastore.mongodb.test.gridfs;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.sql.Blob;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+
+import org.hibernate.Hibernate;
+import org.hibernate.Session;
+import org.hibernate.ogm.datastore.mongodb.impl.MongoDBDatastoreProvider;
+import org.hibernate.ogm.datastore.spi.DatastoreProvider;
+import org.hibernate.ogm.hibernatecore.impl.OgmSessionFactoryImpl;
+import org.hibernate.ogm.utils.TestForIssue;
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+import org.junit.Test;
+
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import com.mongodb.client.gridfs.model.GridFSFile;
+
+/**
+ * @author Davide D'Alto
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ * @see <a href="http://mongodb.github.io/mongo-java-driver/3.4/driver/tutorials/gridfs/">GridFSBucket</a>
+ */
+@TestForIssue(jiraKey = "OGM-786")
+public class GridFSTest extends OgmJpaTestCase {
+
+	static final String BUCKET_NAME = "photos";
+
+	// GridFS is usually for file of size bigger than 16 MB
+	private static final int BLOB_SIZE = 30 * 1024 * 1024; // 30 MB
+
+	private static final byte[] BLOB_CONTENT_1 = createByteArray( 'a', BLOB_SIZE );
+	private static final byte[] BLOB_CONTENT_2 = createByteArray( 'x', BLOB_SIZE );
+	private static final String BLOB_STRING_CONTENT_1 = new String(BLOB_CONTENT_1, StandardCharsets.UTF_8  );
+	private static final String BLOB_STRING_CONTENT_2 = new String(BLOB_CONTENT_2, StandardCharsets.UTF_8  );
+
+	private static final String ENTITY_ID_1 = "photo1";
+	private static final String ENTITY_ID_2 = "photo2";
+
+	private static byte[] createByteArray(char c, int size) {
+		char[] chars = new char[size];
+		Arrays.fill( chars, c );
+		byte[] bytes = new String( chars ).getBytes( StandardCharsets.UTF_8 );
+		return bytes;
+	}
+
+	@Test
+	public void canCreateEntityWithBlob() {
+		inTransaction( em -> {
+			Photo photo = new Photo();
+			photo.setId( ENTITY_ID_1 );
+			photo.setDescription( "photo1" );
+			Blob blob = Hibernate.getLobCreator( em.unwrap( Session.class ) ).createBlob( BLOB_CONTENT_1 );
+			photo.setContentAsBlob( blob );
+			photo.setContentAsByteArray( BLOB_CONTENT_1 );
+			photo.setContentAsString( BLOB_STRING_CONTENT_1 );
+			em.persist( photo );
+		} );
+
+		inTransaction( em -> {
+			MongoDatabase mongoDatabase = getCurrentDB( em );
+			GridFSBucket gridFSFilesBucket = GridFSBuckets.create( mongoDatabase, BUCKET_NAME );
+			MongoCursor<GridFSFile> cursor = gridFSFilesBucket.find().iterator();
+			for ( int i = 0; i < 3; i++ ) {
```

Comment:
I don't understand why you are using a `for` loop in this scenario when you have a cursor.
What happens if the cursor returns 10 values but you only check the first 3?
Also, where the number 3 is coming from?

Reply:
I understood what you mean. Thank for 
 criticism and idea!